traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -0,0 +1,1959 @@
+
+	my $row = $rs_profile->next;
+	if ($row) {
+		my $cdn_name = defined( $row->cdn_id ) ? $row->cdn->name : "";
+
+		$data_obj->{'profile'}->{'name'}   = $row->profile->name;
+		$data_obj->{'profile'}->{'id'}     = $row->profile->id;
+		$data_obj->{'other'}->{'CDN_name'} = $cdn_name;
+
+		%condition = (
+			'profile_parameters.profile' => $data_obj->{'profile'}->{'id'},
+			-or                          => [ 'name' => 'location', 'name' => 'scope' ]
+		);
+		my $rs_config = $self->db->resultset('Parameter')->search( \%condition, { join => 'profile_parameters' } );
+		while ( my $row = $rs_config->next ) {
+			if ( $row->name eq 'location' ) {
+				$data_obj->{'config_files'}->{ $row->config_file }->{'location'} = $row->value;
+			}
+			elsif ( $row->name eq 'scope' ) {
+				$data_obj->{'config_files'}->{ $row->config_file }->{'scope'} = $row->value;
+			}
+		}
+	}
+
+	foreach my $file ( keys %$data_obj->{'config_files'} ) {
+		if ( !defined( $data_obj->{'config_files'}->{$file}->{'scope'} ) ) {
+			$data_obj->{'config_files'}->{$file}->{'scope'} = $self->get_scope($file);
+		}
+	}
+
+	my $file_contents = encode_json($data_obj);
+
+	return $self->render( text => $file_contents, format => 'txt' );
+}
+
+#entry point for server scope api route.
+sub get_server_config {
+	my $self     = shift;
+	my $filename = $self->param("filename");
+	my $id       = $self->param('id');
+	my $scope    = $self->get_scope($filename);
+
+	##check user access
+	if ( !&is_oper($self) ) {
+		return $self->forbidden();
+	}
+
+	##check the scope - is this the correct route?
+	if ( $scope ne 'server' ) {
+		return $self->alert( "Error - incorrect file scope for route used.  Please use the " . $scope . " route." );
+	}
+
+	##verify that a valid server ID has been used
+	my $server_obj = $self->server_data($id);
+	if ( !defined($server_obj) ) {
+		return $self->not_found();
+	}
+
+	#generate the config file using the appropriate function
+	my $file_contents;
+	if ( $filename eq "12M_facts" ) { $file_contents = $self->facts( $server_obj, $filename, $scope ); }
+	elsif ( $filename =~ /to_ext_.*\.config/ ) { $file_contents = $self->to_ext_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "ip_allow.config" ) { $file_contents = $self->ip_allow_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "parent.config" ) { $file_contents = $self->parent_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "records.config" ) { $file_contents = $self->generic_server_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "remap.config" ) { $file_contents = $self->remap_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "hosting.config" ) { $file_contents = $self->hosting_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "cache.config" ) { $file_contents = $self->cache_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "packages" ) {
```

Comment:
how about just another route for this rather than overloading your server configfiles route:

$r->get("/api/$version/server/:id/packages/ats")->over( authenticated => 1 )->to ( 'ApacheTrafficServer#get_server_packages', namespace => 'API::Configs' );